src/backend/aspen/app/views/sample.py

```diff
@@ -447,6 +445,23 @@ def create_sample():
                     user.group.name,
                 ]
 
+            valid_location: Optional[Location] = (
+                g.db_session.query(Location)
+                .filter(Location.id == data["sample"].get("location_id", None))
+                .one_or_none()
+            )
```

Comment:
NIT -- if they send us a location_id, it really is expected to be valid. Let's raise an exception if we can't find one?

Reply:
Long-term, when location_id becomes the primary way to talk about location (rather than current `location`), location_id will go from being optional to being required, right? I assume we'd also rip out accepting text `location` at all then, right?

Reply:
That's the idea!